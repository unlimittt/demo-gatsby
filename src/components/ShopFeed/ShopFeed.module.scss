@import "../../assets/scss/variables";
@import "../../assets/scss/mixins";

.feed {
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  //justify-content: center;

  &__item {
    @include padding(0, 0.4);
    @include margin-bottom(1.25);
    text-align: center;

    &:last-child {
      @include margin-bottom(0.5);
    }

    &-link {
      color: $color-base;

      &:hover,
      &:focus {
        color: $color-base;
      }

      &-title {
        font-size: $typographic-base-font-size;
        text-transform: uppercase;
        font-weight: 600;
        @include line-height(1.5);
        @include margin-top(0);
        @include margin-bottom(0.5);
      }
    }

    &-buy-button {
      display: inline-block;
      height: $button-height;
      padding: 0 24px;
      line-height: $button-height;
      border: 1px solid $color-gray-border;
      text-decoration: none;
      border-radius: $button-border-radius;
      color: $color-base;

      &:hover,
      &:focus {
        color: $color-primary;
      }
    }
  }
}
