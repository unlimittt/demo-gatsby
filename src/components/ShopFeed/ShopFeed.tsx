import React from "react"
import { Link } from "gatsby"
import type { Edges } from "../../types"
import styles from "./ShopFeed.module.scss"
import Img from "gatsby-image"

type Props = {
  edges: Edges
}

const ShopFeed = ({ edges }: Props) => (
  <div className={styles["feed"]}>
    {edges.map(edge => (
      <div className={styles["feed__item"]} key={edge.node.fields.slug}>
        <Link className={styles["feed__item-link"]} to={edge.node.fields.slug}>
          <Img fixed={edge.node.frontmatter.image.childImageSharp.fixed} />
          <h2 className={styles["feed__item-link-title"]}>
            {edge.node.frontmatter.title}
          </h2>
        </Link>
        <Link to="#"
          className={`snipcart-add-item ${styles["feed__item-buy-button"]}`}
          data-item-id={edge.node.fields.slug}
          data-item-price={edge.node.frontmatter.price}
          data-item-url={edge.node.fields.slug}
          data-item-image={edge.node.frontmatter.image.childImageSharp.fixed.src}
          data-item-name={edge.node.frontmatter.title}
        >
          BUY NOW
        </Link>
      </div>
    ))}
  </div>
)

export default ShopFeed
