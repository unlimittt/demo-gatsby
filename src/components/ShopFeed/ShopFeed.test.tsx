import React from "react"
import Enzyme, { shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import ShopFeed from "./ShopFeed"
import allMarkdownRemark from "../../../jest/__fixtures__/shop-all-markdown-remark"

Enzyme.configure({ adapter: new Adapter() })

describe("ShopFeed", () => {
  const props = {
    edges: allMarkdownRemark.allMarkdownRemark.edges
  }

  it("renders correctly", () => {
    const wrapper = shallow(<ShopFeed {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
