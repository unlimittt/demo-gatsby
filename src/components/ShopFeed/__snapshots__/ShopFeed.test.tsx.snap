// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopFeed renders correctly 1`] = `
<div
  className="feed"
>
  <div
    className="feed__item"
    key="/merch/personalized-bottle"
  >
    <mockConstructor
      className="feed__item-link"
      to="/merch/personalized-bottle"
    >
      <Image
        Tag="div"
        alt=""
        durationFadeIn={500}
        fadeIn={true}
        fixed={
          Object {
            "height": 184,
            "src": "/media/customized-coke-bottle.png",
            "srcSet": "",
            "width": 200,
          }
        }
        loading="lazy"
      />
      <h2
        className="feed__item-link-title"
      >
        Personalized Bottle
      </h2>
    </mockConstructor>
    <mockConstructor
      className="snipcart-add-item feed__item-buy-button"
      data-item-id="/merch/personalized-bottle"
      data-item-image="/media/customized-coke-bottle.png"
      data-item-name="Personalized Bottle"
      data-item-price="20.00"
      data-item-url="/merch/personalized-bottle"
      to="#"
    >
      BUY NOW
    </mockConstructor>
  </div>
  <div
    className="feed__item"
    key="/merch/coke-can-sculpted-mug"
  >
    <mockConstructor
      className="feed__item-link"
      to="/merch/coke-can-sculpted-mug"
    >
      <Image
        Tag="div"
        alt=""
        durationFadeIn={500}
        fadeIn={true}
        fixed={
          Object {
            "height": 184,
            "src": "/media/coke_can_sculpted_mug.png",
            "srcSet": "",
            "width": 200,
          }
        }
        loading="lazy"
      />
      <h2
        className="feed__item-link-title"
      >
        Coke Can Sculpted Mug
      </h2>
    </mockConstructor>
    <mockConstructor
      className="snipcart-add-item feed__item-buy-button"
      data-item-id="/merch/coke-can-sculpted-mug"
      data-item-image="/media/coke_can_sculpted_mug.png"
      data-item-name="Coke Can Sculpted Mug"
      data-item-price="40.00"
      data-item-url="/merch/coke-can-sculpted-mug"
      to="#"
    >
      BUY NOW
    </mockConstructor>
  </div>
</div>
`;
